astropy/time/core.py

```diff
@@ -2276,6 +2284,71 @@ def __add__(self, other):
     def __radd__(self, other):
         return self.__add__(other)
 
+    def mean(
+            self: 'Time',
+            axis: Optional[Union[int, Tuple[int]]] = None,
+            dtype: 'npt.DTypeLike' = None,
+            out: Optional['Time'] = None,
+            keepdims: bool = False,
+            *,
+            where: 'npt.NDArray[bool]' = True,
+    ) -> 'Time':
+        """Mean along a given axis.
+
+        This is similar to :meth:`~numpy.ndarray.mean`, but adapted to ensure
+        that the full precision given by the two doubles ``jd1`` and ``jd2``
+        is used, and that corresponding attributes are copied.
+
+        Note that the ``out`` argument is present only for compatibility with
+        ``np.mean``; since `Time` instances are immutable, it is not possible
+        to have an actual ``out`` to store the result in.
+
+        Similarly, the ``dtype`` argument is also present for compatibility with :meth:`~numpy.ndarray.mean`
+        and the ``dtype`` is determined by the ``Time`` format. An :exc:`~ValueError` is raised if this is set to anything
+        other than `None`.
+
+        Note that this implementation may lead to possible loss of precision for instances of ``Time`` spanning many
+        years. Users needing an accurate mean should look to :meth:`~math.fsum` instead.
+        """
+        if dtype is not None:
+            raise ValueError('Cannot set ``dtype`` on `Time` instances')
+        if out is not None:
+            raise ValueError("Since `Time` instances are immutable, ``out`` "
+                             "cannot be set to anything but ``None``.")
+        if self.masked:
+            if where is not None:
+                raise ValueError("The `where` keyword argument cannot be used with instances of MaskedArray")
+
+        kwargs_mean = dict(
+            axis=axis,
+            keepdims=keepdims,
+            where=where,
+        )
+        if axis is None:
+            divisor = self.size
+        else:
+            if axis is int:
```

Comment:
insistence, you are checking if axis is the type, not if it has the type

Reply:
@maxnoe, good catch, I don't know what I was thinking there.